store/sqlstore/user_store.go

```diff
@@ -1942,35 +1943,36 @@ func (us SqlUserStore) DemoteUserToGuest(userID string) (*model.User, error) {
 }
 
 func (us SqlUserStore) AutocompleteUsersInChannel(teamId, channelId, term string, options *model.UserSearchOptions) (*model.UserAutocompleteInChannel, error) {
-	autocomplete := &model.UserAutocompleteInChannel{}
-	uchan := make(chan store.StoreResult, 1)
+	var wg sync.WaitGroup
+	wg.Add(2)
+	var usersInChannel []*model.User
+	var usersInChannelErr error
+	var usersNotInChannel []*model.User
+	var usersNotInChannelErr error
```

Comment:
To further simplify, would it make sense to leverage something like https://github.com/hashicorp/go-multierror and accumulate the errors instead, especially if we don't handle them "specially"?

Reply:
Nice one, I didn't know go-multierror and I didn't know that we were already using it. So... I added it. I still keep the "multierror.Append" after the `wg.Wait()` and not inside the gorouting because I think `multierror.Append` is not thread safe.

Other option is to use a `Group` (https://pkg.go.dev/github.com/hashicorp/go-multierror#Group) of goroutines from the multierror library.

Reply:
Just FYI, coincidentally we're adding `multierror` to `mmctl` for commands that manipulate multiple entities too